.github/CODEOWNERS

```diff
@@ -57,6 +56,7 @@
 /docs/ @ScottTodd
 /samples/ @ScottTodd
 /tools/ @benvanik @GMNGeoffrey
+/tools/dispatch_profiler/ @manishucsd
```

Comment:
> What is the most appropriate home for the dispatch profiler tool?

`benchmarks/` was proposed here: https://github.com/openxla/iree/issues/13464#issuecomment-1540968939

... but we no longer have a top level `benchmarks/` folder as of https://github.com/openxla/iree/pull/13765 (and I'm still confused about where some files live: https://github.com/openxla/iree/pull/13765#discussion_r1204609822)

I think `tools/` makes sense.

Reply:
Sounds good. Since `tools/` makes sense, I am going mark this as resolved.

Reply:
Continuing this thread, from @GMNGeoffrey :
> `build_tools/` really should be `devtools/`, but we've never gotten around to renaming it. I'd prefer you put it there. Underneath `build_tools/` the `build_tools/python/` directory enables a shared root for importing Python packages. What we've done so far is put Python libraries there if they need to be imported from anything other than one directory and then scripts under `build_tools/` add `build_tools/python` to the beginning of their `PYTHONPATH` prior to other imports.
> 
> Everything else under tools/ is C++ and is actually linking in IREE stuff, so I'm -1 on adding it there. Also not a fan of a new top-level `python/`. If we had that at all, I'd expect it for Python bindings for IREE itself, but those are already elsewhere. A subdirectory under `build_tools/benchmarks/` or just directly under `build_tools` seems better to me.

Here's how I see it:

* `tools/` is for things built on top of IREE's runtime and/or compiler components (libraries/binaries). These tools are for developers and users (either to use directly or to fork for their own uses)
* `build_tools/` is for things used to develop the project's implementation details. These should mostly be ignored by users.

Who do we expect to be using this "dispatch profiler"?

Reply:
iree dispatch profiler tool was specifically built with an intention to be used by compiler developers and users for compiling, verifying, and profiling the dispatches they are working on optimizing. Thus, should be in `tools/` directory.

Reply:
> Given that the post-submit workflow has already been verified with the dispatch profiler moved to tools, I would prefer to keep the dispatch_profiler in `iree/tools`.

I don't think this is a good reason to choose a particular location, especially since the entire point of this PR is moving it from one location to another.

> compiling, verifying, and profiling the dispatches they are working on optimizing

These sound a lot like dev workflows to me. The definition of "user" is really fuzzy for IREE, and something we've had confusion about repeatedly. To me, this tool is similar to the Python tools (under `build_tools/benchmarking`) that we have for benchmarking IREE under and not very similar to the C++ tools (under `tools/`) that link IREE itself to provide binary entry points and are built with the IREE Bazel and CMake builds.

I think we should actually step back a bit though because from our meeting this morning, it sounds like you want to expand this to include comparative profiling of other mechanisms for evaluating dispatches, such as Triton. In that case, I think this likely doesn't even belong in this *repo* and would be better placed in the new `openxla-benchmarks` repo, which also aims at creating tools that are compiler agnostic with plugin points for different projects.